### PR TITLE
Fix ModelBlender crash

### PR DESCRIPTION
**Pull request by ZappoMan**
Fix a crash in the ModelBlender logic

### CODE REVIEW

**Comment by ZappoMan** [comment]
Fix this crash - https://app.asana.com/0/26320649018758/102244079680975

**Comment by ZappoMan** [comment]
From what I can tell from this crash (which appeared several times in today's team meeting) -- there was a crash mostly likely related to the Model being deleted while it was being "blended"... I changed the ModelBlender to use std::weak_pointers<> and all other users of the Model to use std::shared_ptr<>

This PR also removes FaceModel which was deprecated but still munging up the code, so I removed that.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0474be59c90969793d49f155bd7ba7510b6b99ec available at: https://gist.github.com/7ca590aa1a5a1a0e607b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7387.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7387.dmg

**Comment by ZappoMan** [comment]
Test Plan:

1) Smoke test of the system
2) Make sure to test avatar "blend shapes" meaning, talking and making sure your avatar mouth is moving.
3) Try to do some multi-user tests, and make sure you can see other people's blend shapes as well.

**Comment by cozza13** [comment]
I do not see the blendshapes on my avatar: https://s3-us-west-1.amazonaws.com/hifi-content/ozan/dev/avatars/hifi_team/chris/_latest/chris.fst

**Comment by ZappoMan** [comment]
@cozza13 - I just ran this build, with your avatar, and I am seeing the mouth move... am I missing something?

**Comment by ZappoMan** [comment]
@cozza13 - I also checked with Ozan and showed him the mouth moving and he said... "that's working... those are his blendshapes"

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2ef5b3727e51c6c31a65a8fa0fdfa1c50e595ff9 available at: https://gist.github.com/7ca590aa1a5a1a0e607b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7387.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7387.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2ef5b3727e51c6c31a65a8fa0fdfa1c50e595ff9 available at: https://gist.github.com/7ca590aa1a5a1a0e607b<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7387.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7387.dmg

File: libraries/avatars/src/AvatarData.cpp [hunk 1]
```diff
@@ -989,18 +988,14 @@ bool AvatarData::hasIdentityChangedAfterParsing(const QByteArray& data) {
     QDataStream packetStream(data);
 
     QUuid avatarUUID;
-    QUrl faceModelURL, skeletonModelURL;
+    QUrl unusedModelURL; // legacy faceModel support
+    QUrl skeletonModelURL;
     QVector<AttachmentData> attachmentData;
     QString displayName;
-    packetStream >> avatarUUID >> faceModelURL >> skeletonModelURL >> attachmentData >> displayName;
+    packetStream >> avatarUUID >> unusedModelURL >> skeletonModelURL >> attachmentData >> displayName;
```
Comment (Atlante45) [inline libraries/avatars/src/AvatarData.cpp]: Why not remove that and bump the protocol?
We were talking about pushing an update, looks like a good opportunity.

Comment (ZappoMan) [inline libraries/avatars/src/AvatarData.cpp]: fair enough

Comment (ZappoMan) [inline libraries/avatars/src/AvatarData.cpp]: after looking into this we determined that the interface doesn't warn on a change to the avatar protocol, so we don't get a recommended upgrade from this idea... so we will leave it backward compatible for now.

**Comment by OmegaHeron** [comment]
Smoke tested Windows 7 - No problems seen.  I also testes using an AV with no blendshapes and that works fine.

**Pull request merged**